graphql_client_codegen/src/query.rs

```diff
@@ -587,34 +587,49 @@ pub(crate) struct UsedTypes {
 }
 
 impl UsedTypes {
+    // We sort the returned inputs and other types in order to ensure that the
+    // generated code is the same every time when given the schema. Without
+    // sorting the order is random, and code generated with the CLI tool may
+    // change even when the source schema does not change.
     pub(crate) fn inputs<'s, 'a: 's>(
         &'s self,
         schema: &'a Schema,
     ) -> impl Iterator<Item = (InputId, &'a StoredInputType)> + 's {
-        schema
+        let mut inputs = schema
             .inputs()
             .filter(move |(id, _input)| self.types.contains(&TypeId::Input(*id)))
+            .collect::<Vec<_>>();
+        inputs.sort_by_key(|(_id, input)| input.name.as_str());
```

Comment:
I think using [`Itertools::sorted_by_key`](https://docs.rs/itertools/latest/itertools/trait.Itertools.html#method.sorted_by_key) can avoid the intermediate `Vec`.